pkg/ccl/changefeedccl/compression.go

```diff
@@ -17,24 +19,110 @@ import (
 	"github.com/klauspost/pgzip"
 )
 
-var useFastGzip = settings.RegisterBoolSetting(
-	settings.ApplicationLevel,
-	"changefeed.fast_gzip.enabled",
-	"use fast gzip implementation",
-	metamorphic.ConstantWithTestBool(
-		"changefeed.fast_gzip.enabled", true,
-	),
-	settings.WithPublic)
+var (
+	useFastGzip = settings.RegisterBoolSetting(
+		settings.ApplicationLevel,
+		"changefeed.fast_gzip.enabled",
+		"use fast gzip implementation",
+		metamorphic.ConstantWithTestBool(
+			"changefeed.fast_gzip.enabled", true,
+		),
+		settings.WithPublic)
+	gzipEncoderPool     = sync.Pool{}
+	fastGzipEncoderPool = sync.Pool{}
+	zstdEncoderPool     = sync.Pool{}
+	gzipDecoderPool     = sync.Pool{}
+	zstdDecoderPool     = sync.Pool{}
+)
 
 type compressionAlgo string
 
-const sinkCompressionGzip compressionAlgo = "gzip"
-const sinkCompressionZstd compressionAlgo = "zstd"
+const (
+	sinkCompressionGzip compressionAlgo = "gzip"
+	sinkCompressionZstd compressionAlgo = "zstd"
+)
 
 func (a compressionAlgo) enabled() bool {
 	return a != ""
 }
 
+type flusher interface {
+	Flush() error
+}
+
+type resettableEncoder interface {
+	Reset(io.Writer)
+}
+
+type resettableDecoder interface {
+	Reset(io.Reader) error
+}
+
+type encoder interface {
+	io.WriteCloser
+	flusher
+	resettableEncoder
+}
+
+type decoder interface {
+	io.ReadCloser
+	resettableDecoder
+}
+
+type encWrapper[T encoder] struct {
```

Comment:
Why does this have to be generic? Can't it just have an `encoder` (the interface type you defined) member?

Reply:
Yeah... My bad. Previously the wrapper was intended to have specific implementations of encoders/decoders. I didn't mention it is useless now

Reply:
Fixed